### PR TITLE
cosmic-applibrary: unstable-2024-01-03 -> 0-unstable-2024-02-09

### PR DESCRIPTION
**Pull request by a-kenji**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/co/cosmic-applibrary/package.nix [hunk 1]
```diff
@@ -1,43 +1,52 @@
...
-  version = "unstable-2024-01-03";
+  version = "0-unstable-2024-02-09";
 
   src = fetchFromGitHub {
     owner = "pop-os";
     repo = pname;
-    rev = "889cb6078c05cdf3eb94f19f64db552fa2be32dc";
-    hash = "sha256-qLAFSHA7YdOWr7ZmLCkQ+aGWb2schANfgdD2BxsrvaE=";
+    rev = "e214e9867876c96b24568d8a45aaca2936269d9b";
+    sha256 = "sha256-fZxDRktiHHmj7X3e5VyJJMO081auOpSMSsBnJdhhtR8=";
```
Comment (kirillrdy) [inline pkgs/by-name/co/cosmic-applibrary/package.nix:20]: from hash to sha256 ? is this because its automated ?

Comment (a-kenji) [reply pkgs/by-name/co/cosmic-applibrary/package.nix]: Good point, I'll try switching it back to hash.
No, I actually got an error with hash itself earlier, but it seems to work now?

**Comment by kirillrdy** [comment]
@a-kenji please consider including a link to diff and/or change log in your commit message

you can run `git log` and grep for `Diff:` to see how other packages do it.

**Comment by kirillrdy** [comment]
sorry cant get it working, but I understand that it shouldn't block this PR
```
 ./results/cosmic-applibrary/bin/cosmic-app-library 
 ERROR cosmic_app_library::app > NoConfigDirectory
Error getting config: com.system76.CosmicTheme.Dark [NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory]

```

**Comment by a-kenji** [comment]
No worries, thanks for the help so far @kirillrdy.

This is also not yet intended to be used standalone, this should serve as a basis for #25964.
And module creation for a cosmic session.

**Comment by wegank** [comment]
@ofborg build cosmic-applibrary cosmic-applibrary.passthru.tests

**Pull request merged**